File: android/app/src/main/java/com/sourish/igblock/RouteBridge.kt
package com.sourish.igblock

import android.util.Log
import android.webkit.JavascriptInterface
import com.sourish.igblock.BuildConfig

class RouteBridge(private val onRoute: (String) -> Unit) {
    @JavascriptInterface
    fun onRouteChanged(path: String) {
        if (BuildConfig.DEBUG) {
            Log.d("RouteBridge", "route changed: $path")
        }
        onRoute(path)
    }
}

File: ios/IGBlock/Sources/IGBlock/RouteBridge.swift
import WebKit

final class RouteBridge: NSObject, WKScriptMessageHandler {
    private let onRoute: (String) -> Void

    init(onRoute: @escaping (String) -> Void) {
        self.onRoute = onRoute
    }

    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
        #if DEBUG
        NSLog("[IGBLOCK-DIAG] RouteBridge received message.body=\(message.body), asString=\(message.body as? String ?? "nil")")
        #endif
        guard let path = message.body as? String else { return }
        onRoute(path)
    }
}
